### PR TITLE
Fix issue return in computed properties & render to match only corret nodes

### LINKED ISSUE #130: Issue in return-in-computed-property & require-render-return
I tried to test new rules in one of my big projects and i have few false positive errors in case when there is CodePath/Function inside with no return.

Affected rules:
- `return-in-computed-property` 
- `require-render-return`

Example of false positive error:
```js
export default {
  computed: {
    foo () {
      const options = []
      this.matches.forEach((match) => {
        options.push(match)
      })
      return options
    }
  }
}
```

### PR DESCRIPTION
**Pull request by armano2**
fixes #130

### CODE REVIEW

**Comment by armano2** [comment]
ok this is still not good :(

File: lib/rules/require-render-return.js [hunk 1]
```diff
@@ -28,7 +28,8 @@ function create (context) {
       forbiddenNodes.forEach(el => {
         if (
           el.loc.start.line >= node.value.loc.start.line &&
-          el.loc.end.line <= node.value.loc.end.line
+          el.loc.end.line <= node.value.loc.end.line &&
+          node.value === el
```
Comment (michalsnik) [inline lib/rules/require-render-return.js]: I think simple:
```
el.loc.start.line === node.value.loc.start.line
```
would be enough, or even just:
```
node.value === el
```
wdyt? (I was sceptic but looks like this kind of comparison works just fine for ASTs)

Plus I think that [this line](https://github.com/armano2/eslint-plugin-vue/blob/ffffa70df6f7bfa2967e048604e917f93c5f9ba2/lib/rules/require-render-return.js#L24)  should be moved to [extecuteOnFunctionsWithoutReturn](https://github.com/armano2/eslint-plugin-vue/blob/ffffa70df6f7bfa2967e048604e917f93c5f9ba2/lib/utils/index.js#L556):

```diff
- if (!isValidReturn()) {
+ if (!isValidReturn() && !node.expression) {
```

File: lib/rules/return-in-computed-property.js [hunk 2]
```diff
@@ -28,7 +28,8 @@ function create (context) {
           if (
             cp.value &&
             el.loc.start.line >= cp.value.loc.start.line &&
-            el.loc.end.line <= cp.value.loc.end.line
+            el.loc.end.line <= cp.value.loc.end.line &&
+            cp.value.parent === el
```
Comment (michalsnik) [inline lib/rules/return-in-computed-property.js]: Same here

File: lib/rules/require-render-return.js (hunk 1 above)
Comment (armano2) [inline lib/rules/require-render-return.js]: @michalsnik thank you for tips 🍡

File: lib/rules/require-render-return.js [hunk 3]
```diff
@@ -20,15 +20,15 @@ function create (context) {
...
+        (item.value.type === 'ArrowFunctionExpression' || item.value.type === 'FunctionExpression')
       )
       if (!node) return
 
       forbiddenNodes.forEach(el => {
         if (
           el.loc.start.line >= node.value.loc.start.line &&
-          el.loc.end.line <= node.value.loc.end.line
+          el.loc.end.line <= node.value.loc.end.line &&
+          el.loc.start.line === node.value.loc.start.line
```
Comment (michalsnik) [inline lib/rules/require-render-return.js]: I meat we can remove all other conditions and leave only this one. Or even just leave:
```
node.value === el
```
:D up to you

Comment (armano2) [inline lib/rules/require-render-return.js]: yup, i need a bit more ☕️

**Pull request merged**